### ISSUE TITLE
23.03.20 AI Society 1주차 머신러닝 수업을 들으면서 궁금했었던 점과 퀴즈에 대한 답변

### ISSUE THREAD

**Issue by jaeyeon1024** [kimdanni/AISociety2#2]
수업을 들으면서 궁금했었던 점과 퀴즈에 대한 답변
Quiz .미분계수가 0인 지점을 찾아 최대 최소를 구하지 않고 왜 경사 하강법을 쓸까?
Ans. 미분계수가 0인 지점을 찾는게 힘든 함수가 존재할 수도 있기 때문에 한 지점에서 미분계수가 0에 근접 할 때 까지 학습률을 고려해서 경사하강법을 쓴다.
Plus Ques 1.  모델 입장에서는 극댓값이나 극솟값이나 둘 다 미분 계수가 0인데 이거에 대한 처리는?(feat. 김다은)
Ans 1. 오차 계산 함수는 차이의 제곱을 하기 때문에 아래로 볼록한 (적어도 가장 오차가 적은 값이 아래에 있는) 함수가 만들어질 것이다.
Plus Ques 2. 학습률이 커서 오차가 0을 계속 지나칠때(+랑- 번갈아가면서)는 학습이 언제 멈추지..? 
Ans 2. 오차가 0이 되면 종료하는거 뿐만 아니라 몇번을 돌리는지나 특정 오차 범위에 들어올 때까지 돌리는 걸로 종료조건을 맞출 수 있을거 같다

Plus Ques 3. 실제로는 종료조건은 어떤 식으로 많이 사용을 하는지? 또한 학습률이나 반복 횟수는 매번 다른건지 정하는 기준이 따로 있는건지?

??

오늘 배운 것에 대한 생각 정리와 궁금한 점을 정리해봤습니다. 틀린부분이나 더 공부해야하는거, 질문에 대한 답변을 듣고 싶습니다.

**Comment by ssam2s** [comment]
Plus Ques 3. 실제로는 종료조건은 어떤 식으로 많이 사용을 하는지? 또한 학습률이나 반복 횟수는 매번 다른건지 정하는 기준이 따로 있는건지?

Ans.
실제 모델 학습에서는 효율적인 학습을 위해 각 프레임워크에서 제공하는 EarlyStopping이라는 조기 종료 함수를 사용해 Validation Loss가 더 이상 줄어들지 않거나(일반적으로는 Overfitting의 경우) 일정 횟수 이상 역으로 증가하는 경우에 학습을 중단하는 방법을 사용함. 외에도 일정 학습 반복 과정에서 Test Accuracy와 같은 측정가능한 척도가 가장 좋은 수치를 보인 best fit 모델을 별도로 저장하기도 함. 학습률(Learning Rate)과 반복 횟수(Epochs)는 다른 하이퍼 파라미터(예를 들면 Batch Size) 또는 모델의 크기나 깊이에 따라 다르게 조정해가면서 씀. 이 조정 과정을 하이퍼 파라미터 튜닝이라고 함. 학습률은 보통 배치 크기와 비례했을때 좋은 성능을 보인다는 말이 있는데 배치 크기와 학습률의 상관 관계에 대한 논문 첨부할테니 읽어보면 도움될듯 !

https://www.sciencedirect.com/science/article/pii/S2405959519303455#fig2
